File: backend/src/Ctor.ReportEngine/ReportEngine/src/main/kotlin/com/ctor/reportEngine/db/entities/Document.kt
package com.ctor.reportEngine.db.entities

import javax.persistence.*

@Entity(name = "Document")
class Document(
    var Name : String,
    var Path : String,
    var Link: String,
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var Id : Long? = null,
) {
    @OneToOne
    @JoinColumn(name = "ProjectDocumentId")
    var ProjectDocument : ProjectDocument? = null
}

File: backend/src/Ctor.Domain/Entities/Document.cs
using Ctor.Domain.Common;

namespace Ctor.Domain.Entities;
public class Document : BaseEntity
{
    public string Name { get; set; }
    public string Path { get; set; }
    public string Link { get; set; }
    public virtual ProjectDocument? ProjectDocument { get; set; }
    public long? ProjectDocumentId { get; set; }

}
